server/controllers/documents/TccGenerate.js

```diff
@@ -0,0 +1,235 @@
+const PDFDocument = require('pdfkit')
+const errors = require('../../../shared/errors')
+const Defense = require('../../models/Defense')
+const Cd = require('../../models/tccdocs/cd')
+const Ata = require('../../models/tccdocs/ata')
+const Certificado = require('../../models/tccdocs/certificado')
+const Credenciamento = require('../../models/tccdocs/credenciamento')
+const Divulgacao = require('../../models/tccdocs/divulgacao')
+
+module.exports = async function generateAllDocs(ctx) {
+  const { id, files } = ctx.params
+  if (
+    ![
+      'ata',
+      'cd',
+      'certificado1',
+      'certificado2',
+      'certificado3',
+      'credenciamento1',
+      'credenciamento2',
+      'credenciamento3',
+      'divulgacao',
+      undefined
+    ].includes(files)
+  ) {
+    ctx.status = 400
+    ctx.body = { param: files, code: errors.INVALID_PARAMS }
+    return
+  }
+  const defenseFind = await Defense.where('id', id).fetch()
+  if (defenseFind === null) {
+    ctx.status = 404
+    ctx.body = { code: errors.NOT_FOUND }
+    return
+  }
+  const defenseDate = defenseFind.get('date')
+  const date = new Date()
+  const dados = {
+    curso: translations[defenseFind.get('course')],
+    tituloTCC: defenseFind.get('title'),
+    nomeDosAlunos: defenseFind.get('students').replace(/, /g, '\n'),
+    tituloOrientador: translations[defenseFind.get('advisorTitle')],
+    orientador: defenseFind.get('advisorName'),
+    tituloCoOrientador: translations[defenseFind.get('coAdvisorTitle')],
+    coOrientador: defenseFind.get('coAdvisorName'),
+    tituloAvaliador1: translations[defenseFind.get('evaluator1Title')],
+    avaliador1: defenseFind.get('evaluator1Name'),
+    tituloAvaliador2: translations[defenseFind.get('evaluator2Title')],
+    avaliador2: defenseFind.get('evaluator2Name'),
+    tituloAvaliador3: translations[defenseFind.get('evaluator3Title')],
+    avaliador3: defenseFind.get('evaluator3Name'),
+    diaDefesa: defenseDate.split('/')[0],
+    mesDefesa: translations[Number(defenseDate.split('/')[1])],
+    anoDefesa: defenseDate.split('/')[2],
+    dia: date.getUTCDate(),
+    mes: translations[date.getUTCMonth()],
+    ano: date.getUTCFullYear(),
+    tituloDiretor: translations.doctor,
+    diretor: 'Josivaldo de Souza Araújo',
+    matricula: defenseFind.get('registrationNumbers').split(', ')[0],
+    discente: defenseFind.get('students').split(', ')[0],
+    palavrasChave: defenseFind.get('keywords'),
+    horarioDefesa: defenseFind.get('time').slice(0, -3),
+    salaDefesa: defenseFind.get('local'),
+    resumo: defenseFind.get('summary'),
+    isDiscente: false,
+    aprovado: !defenseFind.get('passed') ? 'reprovado' : 'aprovado',
+    conceito: 'insuficiente'
+  }
+  if (
+    dados.curso === '' ||
```

Comment:
Vc pode usar o every pra esse teste
`['curso','nome'].every(key => dados[key] !== '')`

Se pelo menos um for igual a `''` retorna `false`